x/community/keeper/proposal_handler.go

```diff
@@ -15,3 +19,62 @@ func HandleCommunityPoolLendDepositProposal(ctx sdk.Context, k Keeper, p *types.
 func HandleCommunityPoolLendWithdrawProposal(ctx sdk.Context, k Keeper, p *types.CommunityPoolLendWithdrawProposal) error {
 	return k.hardKeeper.Withdraw(ctx, k.moduleAddress, p.Amount)
 }
+
+// HandleCommunityPoolProposal is a handler for executing a passed community pool proposal
+func HandleCommunityPoolProposal(ctx sdk.Context, k Keeper, p *types.CommunityPoolProposal) error {
+	logger := k.Logger(ctx)
+
+	// attempt to execute all messages within the passed proposal
+	// Messages may mutate state thus we use a cached context. If one of
+	// the handlers fails, no state mutation is written and the error
+	// message is logged.
+	cacheCtx, writeCache := ctx.CacheContext()
+	msgs, err := p.GetMsgs()
+	if err != nil {
+		return err
+	}
+
+	enabledUrls := k.GetParams(ctx).EnabledProposalMsgUrls
+	for idx, msg := range msgs {
+		if !isMsgEnabled(msg, enabledUrls) {
+			return logAndReturnProposalMsgError(logger, idx, msg, "msg not enabled via params")
+		}
+
+		// assert that the community module account is the only signer of the messages
+		signers := msg.GetSigners()
+		if len(signers) != 1 || !signers[0].Equals(k.moduleAddress) {
+			return logAndReturnProposalMsgError(logger, idx, msg, "invalid signer")
+		}
```

Comment:
still feel like this logic would be more cleanly encapsulated as a method on the msg type, then for each msg we would call a `Validate` method and it can return the appropriate error

(and this is another reason / value from validating the message signers are correct on submission, the earlier we can eject garbage / bad data from the system the more reliable and secure the system will be)

Reply:
this is intentional since the context here matters. This signer validation is only applicable here on the community proposal submission so I think its clearer to do that here. On a msg level, the msg can validate the signer only if the msgs should be ran for a specific signer (which is the case for only the EVMCallMsg).